common/kvstore/src/test/java/org/apache/spark/util/kvstore/DBIteratorSuite.java

```diff
@@ -112,7 +114,8 @@ public void setup() throws Exception {
       t.key = "key" + i;
       t.id = "id" + i;
       t.name = "name" + RND.nextInt(MAX_ENTRIES);
-      t.num = RND.nextInt(MAX_ENTRIES);
+      // Force one item to have an integer value of zero to test the fix for SPARK-23103.
+      t.num = (i != 0) ? (int) RND.nextLong() : 0;
```

Comment:
why the chang from `RND.nextInt(MAX_ENTRIES)`?  this seems fine, just seemed like you were more likely to stress collision on this index before.

Reply:
`nextInt` doesn't return negative values. I can add a mod operator here, but the code already explicitly generates some clashes a few lines below.